src/components/MAPI/utils.ts

```diff
@@ -404,40 +404,60 @@ export async function fetchClusterList(
   auth: IOAuth2Provider,
   provider: PropertiesOf<typeof Providers>,
   namespace?: string,
-  organization?: string
+  organization?: IOrganization
 ): Promise<ClusterList> {
   if (namespace && organization && provider === Providers.AWS) {
-    const [namespacedClusters, nonNamespacedClusters] =
-      await Promise.allSettled([
-        capiv1beta1.getClusterList(httpClientFactory(), auth, { namespace }),
+    const requests = [
+      capiv1beta1.getClusterList(httpClientFactory(), auth, { namespace }),
+    ];
+
+    if (organization?.name) {
+      requests.push(
         capiv1beta1.getClusterList(httpClientFactory(), auth, {
           namespace: 'default',
           labelSelector: {
-            matchingLabels: { [capiv1beta1.labelOrganization]: organization },
+            matchingLabels: {
+              [capiv1beta1.labelOrganization]: organization.name,
+            },
           },
-        }),
-      ]);
-
-    if (
-      namespacedClusters.status === 'rejected' &&
-      nonNamespacedClusters.status === 'rejected'
-    ) {
-      return Promise.reject(nonNamespacedClusters.reason);
+        })
+      );
+    }
+    if (organization?.id) {
+      requests.push(
+        capiv1beta1.getClusterList(httpClientFactory(), auth, {
+          namespace: 'default',
+          labelSelector: {
+            matchingLabels: {
+              [capiv1beta1.labelOrganization]: organization.id,
```

Comment:
This fixes a bug with fetching older AWS clusters that are still in the `default` namespace - we check for matches with both the case-sensitive organization display name (`organization.name`, which has the value of the `ui.giantswarm.io/original-organization-name` annotation) and the organization CR name (`organization.id`)